static/app/views/replays/detail/console/index.tsx

```diff
@@ -30,11 +31,42 @@ function Console({breadcrumbs, startTimestamp = 0}: Props) {
     () =>
       !searchTerm && logLevel.length === 0
         ? breadcrumbs
-        : breadcrumbs.filter(
-            breadcrumb =>
-              breadcrumb.message?.toLowerCase().includes(searchTerm) &&
-              logLevel.includes(breadcrumb.level)
-          ),
+        : breadcrumbs.filter(breadcrumb => {
+            let doesMatch = false;
+            const normalizedSearchTerm = searchTerm.toLowerCase();
+            forIn(breadcrumb.data, (value: any, key: string) => {
+              // the logger key doesn't appear in view so we ignore it.
+              if (doesMatch || key === 'logger') {
+                return;
+              }
+              switch (typeof value) {
+                case 'string':
+                  doesMatch = value.toLowerCase().includes(normalizedSearchTerm);
+                  break;
+                case 'number':
+                  doesMatch = String(value) === normalizedSearchTerm;
+                  break;
+
+                case 'object':
+                  doesMatch = JSON.stringify(value)
+                    .toLowerCase()
+                    .includes(normalizedSearchTerm);
+                  break;
+                default: {
+                  if (Array.isArray(value)) {
+                    doesMatch = value
+                      .join(' ')
+                      .toLowerCase()
+                      .includes(normalizedSearchTerm);
+                  }
+                }
+              }
+            });
```

Comment:
I wasn't sure of how else to manage searching thru record<string, any> so this is what I came up with. Would love to get feedback on a better way to handle it if anyone has any better ideas.

Reply:
can we extract this into a function and add a test for it as well?

Reply:
Yup that works, initially I had it like that but I neglected to check if log level had anything to filter on. So it wasn't working. All good now